web/server/routers/user.ts

```diff
@@ -161,4 +162,35 @@ export const userRouter = router({
         });
       }
     }),
+    searchUsers: procedure
+      .input(
+        z.object({
+          searchParams: z.object({
+            role: z.nativeEnum(Role).optional(),
+            type: z.array(z.nativeEnum(FosterType)).optional(),
+            size: z.array(z.nativeEnum(Size)).optional(),
+            preferredBreeds: z.array(z.nativeEnum(Breed)).optional(),
+            gender: z.array(z.nativeEnum(Gender)).optional(),
+            age: z.array(z.nativeEnum(Age)).optional(),
+            temperament: z.array(z.nativeEnum(Temperament)).optional(),
+            dogsNotGoodWith: z.array(z.nativeEnum(GoodWith)).optional(),
+            medical: z.array(z.nativeEnum(Medical)).optional(),
+            behavioral: z.array(z.nativeEnum(Behavioral)).optional(),
+            houseTrained: z.nativeEnum(Status).optional(),
+            spayNeuterStatus: z.nativeEnum(Status).optional(),
+          }),
+        })
+      )
+      .query(async ({ input }) => {
+        try {
+          const { searchParams } = input;
+          const res = await searchUsers(searchParams);
+          return { data: res };
+        } catch (e) {
```

Comment:
Add conditional to check whether `e` is an instance of `TRPCError` - if it is, throw e, else you can throw the internal server error